evalcache.plugin.zsh

```diff
@@ -7,28 +7,31 @@
 export ZSH_EVALCACHE_DIR=${ZSH_EVALCACHE_DIR:-"$HOME/.zsh-evalcache"}
 
 function _evalcache () {
-  local cmdHash="nohash"
+  local cmdHash="nohash" data="$*" name="$1"
+
+  if typeset -f "${name}" > /dev/null; then
+    data=${data}$(typeset -f "${name}")
+  fi
 
   if builtin command -v md5 > /dev/null; then
-    cmdHash=$(echo -n "$*" | md5)
+    cmdHash=$(echo -n "${data}" | md5)
   elif builtin command -v md5sum > /dev/null; then
-    cmdHash=$(echo -n "$*" | md5sum | cut -d' ' -f1)
+    cmdHash=$(echo -n "${data}" | md5sum | cut -d' ' -f1)
   fi
 
-  local cacheFile="$ZSH_EVALCACHE_DIR/init-${1##*/}-${cmdHash}.sh"
+  local cacheFile="$ZSH_EVALCACHE_DIR/init-${name##*/}-${cmdHash}.sh"
 
   if [ "$ZSH_EVALCACHE_DISABLE" = "true" ]; then
-    eval "$("$@")"
+    eval ${(q)@}
   elif [ -s "$cacheFile" ]; then
     source "$cacheFile"
   else
-    if type "$1" > /dev/null; then
-      (>&2 echo "$1 initialization not cached, caching output of: $*")
+    if type "${name}" > /dev/null; then
+      (>&2 echo "${name} initialization not cached, caching output of: $*")
       mkdir -p "$ZSH_EVALCACHE_DIR"
-      "$@" > "$cacheFile"
-      source "$cacheFile"
+      eval ${(q)@} > "$cacheFile"
```

Comment:
using `eval` to support functions and pre-empt evaluation method in PR #8 (to support in-line environment-variable declarations)

cc: @fmbarina